diff --git a/src/hotspot/share/runtime/safepointMechanism.cpp b/src/hotspot/share/runtime/safepointMechanism.cpp
index 63f237df7..1674d3184 100644
--- a/src/hotspot/share/runtime/safepointMechanism.cpp
+++ b/src/hotspot/share/runtime/safepointMechanism.cpp
@@ -65,7 +65,12 @@ void SafepointMechanism::default_initialize() {
     char* bad_page  = polling_page;
     char* good_page = polling_page + page_size;
 
+#ifdef __MUSL__
+    // OHOS PATCH: avoid SIGSEGV for safepoint polling; readable + 0006 fallback handles it.
+    os::protect_memory(bad_page,  page_size, os::MEM_PROT_READ);
+#else
     os::protect_memory(bad_page,  page_size, os::MEM_PROT_NONE);
+#endif
     os::protect_memory(good_page, page_size, os::MEM_PROT_READ);
 
     log_info(os)("SafePoint Polling address, bad (protected) page:" INTPTR_FORMAT ", good (unprotected) page:" INTPTR_FORMAT, p2i(bad_page), p2i(good_page));
